executor/compiler.go

```diff
@@ -122,7 +122,116 @@ func CountStmtNode(stmtNode ast.StmtNode, inRestrictedSQL bool) {
 	if inRestrictedSQL {
 		return
 	}
-	metrics.StmtNodeCounter.WithLabelValues(GetStmtLabel(stmtNode)).Inc()
+
+	typeLabel := GetStmtLabel(stmtNode)
+	metrics.StmtNodeCounter.WithLabelValues(typeLabel).Inc()
+
+	if !config.GetGlobalConfig().Status.RecordQPSbyDB {
+		return
+	}
+
+	dbLabels := getStmtDbLabel(stmtNode)
+	for _, dbLabel := range dbLabels {
+		metrics.DbStmtNodeCounter.WithLabelValues(dbLabel, typeLabel).Inc()
+	}
+}
+
+func getStmtDbLabel(stmtNode ast.StmtNode) []string {
+	dbLabelMap := make(map[string]int)
```

Comment:
maybe better use `dbLabelSet := make(map[string]struct{})`(empty struct alloc zero), and remove L189~193's copy  and return `make(map[string]struct{})` directly?

Reply:
ok